bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/DeviceInspector.java

```diff
@@ -102,15 +104,128 @@ private Map<String, String> readDeviceProperties(IndividualAddress address) {
+                                MAX_ROUTED_APDU_LENGTH, 1, 1, false, OPERATION_TIMEOUT);
+                        if (result != null) {
+                            maxApdu = Integer.toString(toUnsigned(result));
+                        }
+                    } catch (Exception e) {
+                        // allowed to fail, optional
+                    }
+                    if (!maxApdu.isEmpty()) {
+                        logger.trace("Max APDU of device {} is {} bytes", address, maxApdu);
+                    } else {
+                        logger.trace("Max APDU of device {} not set, fallback to 14 bytes", address);
+                        maxApdu = "14"; // see spec
+                    }
+                }
+
+                Thread.sleep(OPERATION_INTERVAL);
+                byte[] orderInfo = getClient().readDeviceProperties(address, DEVICE_OBJECT, PID.ORDER_INFO, 1, 1, false,
+                        OPERATION_TIMEOUT);
+                if (orderInfo != null) {
+                    final String hexString = toHex(orderInfo, "");
+                    if ((!"ffffffffffffffffffff".equals(hexString)) && (!"00000000000000000000".equals(hexString))) {
+                        String result = new String(orderInfo);
+                        result = result.trim();
+                        if (result.isEmpty()) {
+                            result = "0x" + hexString;
+                        } else {
+                            final String printable = result.replaceAll("[^\\x20-\\x7E]", ".");
+                            if (!printable.equals(result)) {
+                                result = printable + " (0x" + toHex(orderInfo, "") + ")";
+                            }
+                        }
+                        logger.trace("Order code of device {} is \"{}\"", address, result);
+                        ret.put(MANUFACTURER_ORDER_INFO, result);
+                    }
+                }
+
+                // read FRIENDLY_NAME, optional
+                Thread.sleep(OPERATION_INTERVAL);
+                try {
+                    byte[] count = getClient().readDeviceProperties(address, ROUTER_OBJECT, PID.FRIENDLY_NAME, 0, 1,
+                            false, OPERATION_TIMEOUT);
+                    if ((count != null) && (toUnsigned(count) == 30)) {
+                        StringBuffer buf = new StringBuffer(30);
+                        for (int i = 1; i <= 30; i++) {
+                            Thread.sleep(OPERATION_INTERVAL);
+                            // for some reason, reading more than one character per message fails
+                            // reading only one character is inefficient, but works
+                            byte[] data = getClient().readDeviceProperties(address, ROUTER_OBJECT, PID.FRIENDLY_NAME, i,
+                                    1, false, OPERATION_TIMEOUT);
+                            if (toUnsigned(data) != 0) {
+                                if (data != null) {
```

Comment:
Looking at `toUnsigned`, it will return 0 if the parameter is null. So this test is useless as already covered by the previous test.

Reply:
right, but compiler does not get it, resulting in a compile error without the check,,,,